### ISSUE TITLE
[QUESTION] Substituting a species for another within HEMCO  

### ISSUE THREAD

**Issue by arianatribby** [geoschem/geos-chem#383]
I am interested in replacing propane (C3H8) emissions with ethane (C2H6) emissions in HEMCO. I was thinking of simply replacing the source file and source variable of propane with ethane for all of the base emissions under the base emissions section in the HEMCO_Config.rc file. Then, I would introduce a scale factor in the section 'Settings' of HEMCO_Config.rc using ```EmisScale_<SpeciesName>```. 

For example, under XIAO base emissions, here are the emissions for C3H8: 

```
0 XIAO_C3H8    $ROOT/XIAO/v2014-09/C3H8_C2H6_ngas.geos.1x1.nc C3H8         1985/1/1/0 C xy kgC/m2/s C3H8 6/7/26/22 1 5

```
I was thinking of replacing that line with this instead: 

```
0 XIAO_C3H8    $ROOT/XIAO/v2014-09/C3H8_C2H6_ngas.geos.1x1.nc C2H6         1985/1/1/0 C xy kgC/m2/s C3H8 6/7/26/22 1 5

```

However, if I were to repeat this for all of the base emissions of C3H8 (GEIA, CEDS, XIAO), I run into a problem because C2H6 has an additional base emission that doesn't include C3H8. So by using this method, I would miss the following C2H6 emissions:

```
#==============================================================================
# --- 2010 Global fossil fuel and biofuel emissions of C2H6 (Tzompa-Sosa et al. (2017) ---
#==============================================================================
(((C2H6_2010
0 C2H6_2010_oilgas   $ROOT/C2H6_2010/v2017-05/C2H6_global_anth_biof.2010$MM.nc ANTHR_C2H6   2010/1-12/1/0 C xy kgC/m2/s C2H6 - 1 100
0 C2H6_2010_biofuel  $ROOT/C2H6_2010/v2017-05/C2H6_global_anth_biof.2010$MM.nc BIOFUEL_C2H6 2010/1-12/1/0 C xy kgC/m2/s C2H6 - 2 100
)))C2H6_2010

```

Is there a straightforward way to do this? Also, the units for Tzompa-Sosa, XIAO inventories are kgC/m2/s, but CEDS/GEIA are kg/m2/s. In constructing my scaling factor to relate C2H6 to C3H8 emissions, should I consider their relationship in terms of just mass or kgC? 

Note: I run with GEOS-Chem Classic version 12.1.1, 4x5 with merra2, but I am trying this with HEMCO Standalone with geosfp 4x5 version 12.1.1.

**Comment by yantosca** [comment]
Thanks for writing.  What might be easier is to run HEMCO in a GEOS-Chem or standalone simulation, to save out the C2H6 emissions to the HEMCO_diagnostics*.nc files.  Then you can read the C2H6 emissions back from the HEMCO_diagnostics files and apply the relevant scale factors.  In the HEMCO_Diagn.rc file the diagnostic output should be converted from kgC/m2/s to kg/m2/s.

We are planning on removing implicit unit conversions in HEMCO, so that all scale factors will be explicitly contained in the HEMCO_Config.rc file in a future version such as 13.0.0 or 13.1.0.

**Comment by arianatribby** [comment]
Thanks for your help. Yes, this is the process I've started to work on. When HEMCO C3H8 reads in the C2H6 nc4 emissions files, wouldn't it be reading each individual type of emission from the $ROOT path? So the emissions would be separated into anthropogenic vs biogenic etc, and I would run into the same issue unless I created a new category of C3H8 emissions under Tzompa-Sosa? As for the units, I'm understanding that the input file that HEMCO should read for new C3H8 should be in units of kg/m2/s and the best way to do this would be to specify that on HEMCO_sa_Diagn.rc when the nc4 file is generated from standalone. Is this correct?

**Comment by arianatribby** [comment]
I generated emissions with the Standalone. What is the best way to read these emissions from the diagnostic nc file into the HEMCO_Config.rc file? I was thinking of scaling all C3H8 emissions to zero, introducing a new collection (which would be the total C2H6 emissions from the diagnostic file), and apply a global mask so that it would substitute that collection in. Is this what you would recommend?

**Comment by msulprizio** [comment]
We would recommend create a new "inventory" in HEMCO_Config.rc, and adding a line telling HEMCO where to find your file and what variable(s) to read from that file. If you want the emissions that you generated to overwrite all other emissions for that species and emission sector, you will just need to give them a higher hierarchy in HEMCO_Config.rc. The hierarchy is the last value in the HEMCO entry. See this section of the HEMCO user's guide for more details: http://wiki.seas.harvard.edu/geos-chem/index.php/The_HEMCO_User%27s_Guide#Base_emissions.

**Comment by msulprizio** [comment]
To read emissions for another species (C2H6) into C3H8 emissions. You can just tell HEMCO to read the C2H6 variable from the file (`sourceVar` column) and input that into species C3H8 (`Species` column), making sure the hierarchy for that new entry is higher than that for other C3H8 entries. For reference, HEMCO entries are as follows:

`# ExtNr Name sourceFile sourceVar sourceTime C/R/E SrcDim SrcUnit Species ScalIDs Cat Hier`

**Comment by msulprizio** [comment]
I think I see the issue. You define a new inventory `AT_C3H8` but never set it to true. At the top of HEMCO_Config.rc, add a line to the extension switches section:

```
# ExtNr ExtName                on/off  Species
0       Base                   : on    *
    --> GC_RESTART             :       true	
    --> HEMCO_RESTART          :       true
    --> AEIC                   :       true
    --> BIOFUEL                :       true
    --> BOND                   :       true
    --> BRAVO                  :       true
    --> APEI                   :       true
    --> XIAO                   :       false
    --> AT_C3H8                :       true    # <-- ADD YOUR INVENTORY HERE
    --> C2H6_2010              :       true

```

**Comment by arianatribby** [comment]
Thanks for your help. I fixed the issue you brought up and re-ran a standard chemistry run, but it still doesn't appear to be working. I plotted the difference in emissions from the old and new diagnostic files and it is still zero everywhere. If there were something wrong with the input file, wouldn't it come up in the HEMCO.log file? I searched for C3H8 but no errors came up. If you don't mind taking another look at my updated HEMCO.Config, that would be so helpful. Thanks!
[HEMCO_Config_8_26.txt](https://github.com/geoschem/geos-chem/files/5134110/HEMCO_Config_8_26.txt)

**Comment by msulprizio** [comment]
Could you try setting Verbose and Warnings to 3 at the top of your HEMCO_Config.rc? That should provide more information about the handling of your new emissions in the HEMCO.log file. Once you rerun, please attach your HEMCO.log file here and I can have a closer look as well.

**Comment by arianatribby** [comment]
Thanks, I implemented your suggestions. You can find the log file (~ 4 GB) at the box link below. 

With a quick browse through the file, I didn't see anything alarming yet. Other emissions containers were skipped because of the hierarchy that I implemented (eg "Skip container AEIC_C3H8 because of AT_air_C3H8"), which was encouraging.  Also, the "Container AT_air_C3H8 Array" sum did not look to be nonzero.  Since the log file is so large, it is hard for me to look through all of it - if you know what section I should look into more closely, I would really appreciate it. Thanks for your time!

https://caltech.box.com/s/opfjrg22r32kl3e8gmavu3kjyo8fuann

**Comment by arianatribby** [comment]
Hi, I just wanted to follow up. I took a look at the GEOSChem.SpeciesConc files, and found that the emissions were in fact incorporated: 

Here is C3H8 concentration resulting from old emission files: 
<img width="929" alt="Screen Shot 2020-10-05 at 3 06 34 PM" src="https://user-images.githubusercontent.com/56133461/95137785-b3632f80-071d-11eb-9cef-700d78031bb9.png">

C3H8 concentration resulting from new emissions:
<img width="968" alt="Screen Shot 2020-10-05 at 3 03 35 PM" src="https://user-images.githubusercontent.com/56133461/95137741-9b8bab80-071d-11eb-8385-8f915c5b94d9.png">

The spatial pattern is what I was expecting to see with the new emissions. Looks great! However, the HEMCO diagnostic file still shows no difference between the old and new emissions: 

```
atom4_emissions = xr.open_mfdataset('/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_04_14_20/HEMCO_diagnostics.2018*.nc.mp')

atom4_newc3h8emis = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8emiss_9_24_20/HEMCO_diagnostics.2018*")

plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
(atom4_emissions_avg.EmisC3H8_Total.sum(dim='lev') - atom4_newc3h8emis_avg.EmisC3H8_Total.sum(dim='lev')).plot()
ax1.coastlines()
plt.show()
```
![thumbnail_image](https://user-images.githubusercontent.com/56133461/95138777-ec040880-071f-11eb-8a62-21ff2422aa6f.png)

(I also did an argmax for the difference and it is zero.)


Is it possible that the emissions are incorporated but a particular file is used to generate the diagnostics? I don't see any clues in the HEMCO log or the diagnostic file. Here is the Diagn file associated with that run: 

[HEMCO_Diagn.rc.txt](https://github.com/geoschem/geos-chem/files/5330436/HEMCO_Diagn.rc.txt)

I need the updated emissions output to continue. Thanks for your help!

**Comment by yantosca** [comment]
Hi @arianatribby, thanks for writing.  I wonder if the problem is this line:
```
(atom4_emissions_avg.EmisC3H8_Total.sum(dim='lev') - atom4_newc3h8emis_avg.EmisC3H8_Total.sum(dim='lev')).plot()
```
In the code snippet above you are reading in the data to datasets atom4_emissions and atom4_newc3h8emis but you are plotting atom4_emissions_avg and atomv_newc3h8emis_avg.  I don't know if you omitted other code lines but double check to see if this plot line is not in error.

**Comment by msulprizio** [comment]
Also, have you tried looking at the other C3H8 diagnostics to see if the differences are seen there? From your HEMCO_Config.rc file, you added:

```
#===========================================
# ----- AT EMISSIONS (from HEMCO Standalone)
#===========================================
(((AT_C3H8
0 AT_ship_C3H8     /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Ship $YYYY/$MM/1/0 C xy kg/m2/s C3H8 - 10 100
0 AT_bioburn_C3H8  /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_BioBurn $YYYY/$MM/1/0 C xy kg/m2/s C3H8 - 5 100
0 AT_anthro_C3H8   /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Anthro $YYYY/$MM/1/0 C xyz kg/m2/s C3H8 - 1 100
0 AT_air_C3H8      /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Aircraft $YYYY/$MM/1/0 C xyz kg/m2/s C3H8 - 20 100
)))AT_C3H8
```

In HEMCO_Diagn.rc, try saving out:

```
EmisC3H8_Total     C3H8   -1   -1  -1   3   kg/m2/s  C3H8_total_emissions
EmisC3H8_Anthro    C3H8   0    1   -1   2   kg/m2/s  C3H8_emissions_from_anthropogenic
EmisC3H8_BioBurn   C3H8   0    5   -1   2   kg/m2/s  C3H8_emissions_from_biomass_burning
EmisC3H8_Air       C3H8   0    20  -1   3   kg/m2/s  C3H8_emissions_from_airplanes
```

Here, I (1) changed the units back to the original emissions unit to make sure the unit conversions in HEMCO to/from units carbon aren't the issue; (2) changed the BioBurn entry to reflect that the emissions are not from the GFED extension but base emissions instead; and (3) added the Air entry. Since the total and airplane output are 3D, take care to ensure you're looking at the total column differences or differences at the same level in your plots.

**Comment by arianatribby** [comment]
Thanks for writing. I've isolated the problem. Sorry for not including all of the code. There are two diagnostic files that are outputted (one per month, and I am looking at a two-month period), so I first get the time average for the new and old period, then compare them.

```
atom4_emissions = xr.open_mfdataset('/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_04_14_20/HEMCO_diagnostics.2018*.nc.mp')
atom4_emissions_avg = atom4_emissions.mean(dim='time')

atom4_newc3h8emis = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8emiss_8_26_20/HEMCO_diagnostics.2018*")
atom4_newc3h8emis_avg = atom4_emissions.mean(dim='time')


plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
(atom4_emissions_avg.EmisC3H8_Total.sum(dim='lev') - atom4_newc3h8emis_avg.EmisC3H8_Total.sum(dim='lev')).plot()
ax1.coastlines()
plt.show()

```
![image](https://user-images.githubusercontent.com/56133461/95236706-1d321680-07bc-11eb-8c7a-2611bbb4a912.png)


So I am comparing the differences at the same level, but the problem is using the time average. Here is a plot of the time-averaged new emissions of c3h8: 
```
plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
atom4_newc3h8emis_avg.EmisC3H8_Total.sum(dim='lev').plot()
ax1.coastlines()
plt.show()
```
![image](https://user-images.githubusercontent.com/56133461/95235933-1525a700-07bb-11eb-9098-63f4ae961138.png)

Which looks like the old emissions. The plots of the individual files of new emissions of c3h8, not time averaged, look like I would expect: 
```
atom4_newc3h8emis_file1 = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8emiss_8_26_20/HEMCO_diagnostics.201804010000.nc.mp")
atom4_newc3h8emis_file2 = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8emiss_8_26_20/HEMCO_diagnostics.201805010000.nc.mp")
```

```
plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
atom4_newc3h8emis_file1.EmisC3H8_Total.sum(dim='lev').plot()
ax1.coastlines()
plt.show()
```
![image](https://user-images.githubusercontent.com/56133461/95236162-6cc41280-07bb-11eb-973c-8b44e9a91b1a.png)

```
plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
atom4_newc3h8emis_file2.EmisC3H8_Total.sum(dim='lev').plot()
ax1.coastlines()
plt.show()
```
![image](https://user-images.githubusercontent.com/56133461/95236173-73528a00-07bb-11eb-8be7-882b996829bf.png)

And the new emissions are definitely different compared to the old, file by file: 
```
atom4_emissions_file1 = xr.open_mfdataset('/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_04_14_20/HEMCO_diagnostics.201804010000.nc.mp')
atom4_emissions_file2 = xr.open_mfdataset('/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_04_14_20/HEMCO_diagnostics.201805010000.nc.mp')
```
```
plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
(atom4_newc3h8emis_file1.EmisC3H8_Total.sum(dim='lev') - atom4_emissions_file1.EmisC3H8_Total).sum(dim='lev').plot()
ax1.coastlines()
plt.show()
```
![image](https://user-images.githubusercontent.com/56133461/95237692-62a31380-07bd-11eb-8765-725f3f3a3d31.png)

```
plt.figure(figsize=(10,6))
ax1 = plt.subplot(1,1,1, projection=ccrs.PlateCarree())
(atom4_newc3h8emis_file2.EmisC3H8_Total.sum(dim='lev') - atom4_emissions_file2.EmisC3H8_Total).sum(dim='lev').plot()
ax1.coastlines()
plt.show()
```
![image](https://user-images.githubusercontent.com/56133461/95237713-68005e00-07bd-11eb-9558-aaae3990ed56.png)


So I now understand that the issue was with using the time average to compare the campaign period all at once. I could maybe just compare sums of that time period instead of averages. What would you recommend?


As for the Diagn file, thanks for including that code. I was confused about where the category number was coming from, as it didn't match the one from the HEMCO_log/Config file (but now it does match both of those files with your code). Just to make sure, the other output in the diagnostic file (anthro, bioburn, etc) doesn't impact the total emissions output, right? So if I were to exclude ship from the diagn file, it wouldn't impact the value in the total category?

**Comment by yantosca** [comment]
Thanks @arianatribby for letting us know that the diagnostic output is OK.

I personally like to convert the data in a xr.DataArray object to a numpy ndarray, so that I can use the numpy functions.   For example:

```python
refarray = atom4_newc3h8emis_file2["EmisC3H8_Total].values
refmean = np.nanmean(refarray, axis=1)
```

You can use [numpy.mean](https://numpy.org/doc/stable/reference/generated/numpy.mean.html) or [numpy.nanmean](https://numpy.org/doc/stable/reference/generated/numpy.nanmean.html).  The nanmean function will ignore NaN values, which are often used to represent missing data.

The total emissions per species are computed by HEMCO over all categories (-1)
```
EmisC3H8_Total     C3H8   -1   -1  -1   3   kg/m2/s  C3H8_total_emissions
```
So they are consistent with whatever emissions you specify in the HEMCO_Config.rc file.

**Comment by arianatribby** [comment]
Thanks, Bob. I wanted to test whether the emissions were indeed the same for C3H8 and C2H6. I ran another standard run, this time not only substituting C2h6 for C3H8, but also overriding old C2H6 emissions with new C2h6 from HEMCO 3.0 standalone file. I've attached the HEMCO_Config file. Here are the results, where I plot two monthly diagnostic files (in units of kg/m2/s): 

```
atom4_newc3h8emis_file1 = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8_C2H6emiss_10_8_20/HEMCO_diagnostics.201804010000.nc.mp")
atom4_newc3h8emis_file2 = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8_C2H6emiss_10_8_20/HEMCO_diagnostics.201805010000.nc.mp")

atom4_newc3h8emis_file1_c2h6_kgm2s = atom4_newc3h8emis_file1['EmisC2H6_Total'].sum(dim='lev').values.flatten()
atom4_newc3h8emis_file1_c3h8_kgm2s = atom4_newc3h8emis_file1['EmisC3H8_Total'].sum(dim='lev').values.flatten()

atom4_newc3h8emis_file2_c2h6_kgm2s = [atom4_newc3h8emis_file2['EmisC2H6_Total'].sum(dim='lev').values.flatten()]
atom4_newc3h8emis_file2_c3h8_kgm2s = atom4_newc3h8emis_file2['EmisC3H8_Total'].sum(dim='lev').values.flatten()
```

```
hv.Points(
    (atom4_newc3h8emis_file1_c2h6_kgm2s, 
     atom4_newc3h8emis_file1_c3h8_kgm2s)
).opts(
    height=400,
    width=400,
    padding=.1,
    xlabel='c2h6 emissions',
    ylabel='c3h8 emissions'
)
```
<img width="383" alt="Screen Shot 2020-10-08 at 3 00 38 PM" src="https://user-images.githubusercontent.com/56133461/95520656-a5f7b080-097c-11eb-8844-6cf240b17009.png">

```
hv.Points(
    (atom4_newc3h8emis_file2_c2h6_kgm2s, 
     atom4_newc3h8emis_file2_c3h8_kgm2s)
).opts(
    height=400,
    width=400,
    padding=.1,
    xlabel='c2h6 emissions',
    ylabel='c3h8 emissions'
)
```
<img width="377" alt="Screen Shot 2020-10-08 at 3 00 47 PM" src="https://user-images.githubusercontent.com/56133461/95520673-adb75500-097c-11eb-9560-36e67a2f4b3d.png">

The units from the diagnostic file were kg/m2/s. I would expect this to be 1:1 since I replaced C3H8 with C2H6? The shape looks the same when I convert to mol.

In the new config file, the new emissions categories are AT_C3H8 and AT_C2H6, which I override using the hierarchy techniques discussed in this issue. 
[HEMCO_Config_10_8_20.rc.txt](https://github.com/geoschem/geos-chem/files/5351617/HEMCO_Config_10_8_20.rc.txt)

**Comment by arianatribby** [comment]
Just a follow up from my previous post. I am expecting the outputted emissions of c3h8/c2h6 to be 1:1, and for the original emissions from the HEMCO standalone diagnostics to be very different from each other if HEMCO has incorporated them correctly. Here is the correlation between c3h8 and c2h6 from the raw standalone:
 
<img width="503" alt="Screen Shot 2020-10-09 at 4 02 55 PM" src="https://user-images.githubusercontent.com/56133461/95639727-71542980-0a4e-11eb-86c8-5792719f826a.png">

This is good - the plots in my previous post show the correlation is better, so C3H8 is being substituted. But I am not sure what is causing the deviation from the 1:1 (see previous post)?

Besides that question, I am also wondering - shouldn't I expect the C2H6 emissions from the HEMCO standalone to be equal to the emissions outputted by the diagnostic file after a standard chemistry run (since the standalone emissions are what it read in)? They are not the same.

For reference, here I read in the standalone diagnostic files that I use for input into the Config file for c2h6/c3h8 emissions for a standard chemistry run:
```
atom4_sa_file1 = xr.open_mfdataset("/home/atribby/GEOS-Chem/rundirs/hemco_v3.0_standalone/atom4_time_period/HEMCO_sa_diagnostics.201804010000.nc")
atom4_sa_file1_c2h6_kgm2s = atom4_sa_file1['EmisC2H6_Total'].sum(dim='lev').values.flatten()
```
Here, I read in the c2h6/c3h8 from the diagnostic files that were outputted along with the species concentrations after a standard chemistry run:
```
atom4_newc3h8emis_file1 = xr.open_mfdataset("/central/groups/POW/TemporaryStorage/atribby/merra_4x5_standard/atom4_newC3H8_C2H6emiss_10_8_20/HEMCO_diagnostics.201804010000.nc.mp")
atom4_newc3h8emis_file1_c2h6_kgm2s = atom4_newc3h8emis_file1['EmisC2H6_Total'].sum(dim='lev').values.flatten()
```
Plot c2h6 from both files (comparing a monthly output). They are different.
```
hv.Points(
    (atom4_sa_file1_c2h6_kgm2s,
     atom4_newc3h8emis_file1_c2h6_kgm2s)
).opts(
    height=500,
    width=500,
    padding=.1,
    xlabel='c2h6 SA diagnostics',
    ylabel='c2h6 Run diagnostics'
)
```
<img width="524" alt="Screen Shot 2020-10-09 at 4 10 02 PM" src="https://user-images.githubusercontent.com/56133461/95638860-0bb26e00-0a4b-11eb-8aa8-2aaed856c76c.png">


Shouldn't the C2H6 emissions from the standalone and the diagnostic files after the run be the same? Both files are monthly output during the same time period. Where in the log file can I double check that other emissions categories are being overwritten, or do you have another suggestion?

**Comment by msulprizio** [comment]
I wonder if this had to do with the conversion between units species and units carbon. Both C2H6 and C3H8 are emitted as kgC/m2/s by default in HEMCO. When the input units in HEMCO are in kg/m2/s, HEMCO will automatically convert the units of those species to kgC/m2/s. See [this section of the HEMCO users guide](http://wiki.seas.harvard.edu/geos-chem/index.php/The_HEMCO_User's_Guide#Units_in_HEMCO) for more details.

**Comment by arianatribby** [comment]
Thanks for writing. I understand that HEMCO converts to kgC/m2/s, so I requested the output of the diagnostic files be in units of kg. Here is what I wrote in the  diagnostic file:

```
EmisC2H6_Total     C2H6   -1     -1  -1   3   kg/m2/s   C2H6_total_emissions
EmisC3H8_Total     C3H8   -1     -1  -1   3   kg/m2/s   C3H8_total_emissions
```
So I didn't do any conversions by hand. Was this sufficient? I also converted back to kgC units for comparison, but the plots look the same: 

```
hv.Points(
    (atom4_newc3h8emis_file1_c2h6_kgm2s*(12/33.07)*2, 
     atom4_newc3h8emis_file1_c3h8_kgm2s*(12/44.1)*3)
).opts(
    height=500,
    width=500,
    padding=.1,
    xlabel='c2h6 emissions',
    ylabel='c3h8 emissions'
)
```
<img width="519" alt="Screen Shot 2020-10-13 at 10 11 29 AM" src="https://user-images.githubusercontent.com/56133461/95893206-7d850300-0d3c-11eb-8eb6-d73cc520dd28.png">

Maybe there is a different issue? Sorry if I am still misunderstanding. Thanks for your time and help.

**Comment by msulprizio** [comment]
I'll try to explain my reasoning. Your input data (AT_*) are in units kg/m2/s. HEMCO will read those fields and immediately convert to the units expected by HEMCO, which are kgC/m2/s for VOCs like C2H6 and C3H8. When reading the same values into species C2H6 and C3H8, HEMCO will use different scaling to convert from kg/m2/s to kgC/m2/s. For C2H6, the following values will be used:

```
MW_g: 30.08
EmMW_g: 12.0
MolecRatio: 2.0
```

while C3H8 will use:

```
MW_g: 44.11
EmMW_g: 12.0
MolecRatio: 3.0
```

Any scale factors or masks applied to those emissions will be applied to the values converted to kgC/m2/s (which will differ for C2H6 and C3H8). The fluxes in kgC/ms/2 will be passed to GEOS-Chem. Finally, to save the emissions diagnostics as specified in your HEMCO_Diagn.rc, HEMCO will convert from kgC/m2/s back to kg/m2/s using the above values, but since scale factors and masks were applied to the differing values in kgC/m2/s, you will not get 1:1 results. Does this make sense?

**Comment by arianatribby** [comment]
Yes, it is starting to make more sense why they could be different. However, I didn't specify any scale factors or masks for C3H8 and C2H6 in the config file. Here are the corresponding entries: 

```
(((AT_C3H8
0 AT_ship_C3H8    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Ship $YYYY/$MM/1/0 C xy kg/m2/s C3H8 - 10 100
0 AT_bioburn_C3H8    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_BioBurn $YYYY/$MM/1/0 C xy kg/m2/s C3H8 - 5 100
0 AT_anthro_C3H8    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Anthro $YYYY/$MM/1/0 C xyz kg/m2/s C3H8 - 1 100
0 AT_air_C3H8    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Aircraft $YYYY/$MM/1/0 C xyz kg/m2/s C3H8 - 20 100
)))AT_C3H8


(((AT_C2H6
0 AT_ship_C2H6    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Ship $YYYY/$MM/1/0 C xy kg/m2/s C2H6 - 10 100
0 AT_bioburn_C2H6    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_BioBurn $YYYY/$MM/1/0 C xy kg/m2/s C2H6 - 5 100
0 AT_anthro_C2H6    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Anthro $YYYY/$MM/1/0 C xyz kg/m2/s C2H6 - 1 100
0 AT_air_C2H6    /central/scratchio/atribby/gc_rundirs/merra2_4x5_standard/HEMCO_sa_diagnostics.$YYYY$MM010000.nc EmisC2H6_Aircraft $YYYY/$MM/1/0 C xyz kg/m2/s C2H6 - 20 100
)))AT_C2H6
```
In any case, all I want to do is to check to make sure c2h6 matches c3h8 emissions, and that the new emissions were incorporated correctly. What might be a better way to double check?

**Comment by msulprizio** [comment]
>Thanks for writing. I understand that HEMCO converts to kgC/m2/s, so I requested the output of the diagnostic files be in units of kg. Here is what I wrote in the diagnostic file:

We recently discovered that HEMCO does not convert from kgC back to kg species in GEOS-Chem 12.9.0 and earlier. This means even if units in HEMCO output are listed as kg/m2/s for VOC species traditionally carried in GEOS-Chem in units carbons, the units are actually kgC/m2/s. This may explain the differences you're seeing between C2H6 and C3H8 emissions. This will be fixed in 13.0.0 with the [removal of carbon-based units](https://github.com/geoschem/geos-chem/issues/309).

**Comment by msulprizio** [comment]
I will close out this issue for now, but please contact us to reopen or feel free to submit a new issue issue if there are any remaining questions.
